src/functions-reference/higher-order_functions.Rmd

```diff
@@ -412,7 +412,129 @@ the same size:
+
+Stan provides two higher order functions for solving initial value
+problems specified as Differential-Algebraic Equations (DAEs) with index-1 [@serban_user:2021].
+
+Solving an initial value DAE means given a set of residual functions
+$r(y'(t, \theta), y(t, \theta), t)$ and initial conditions $(y(t_0, \theta), y'(t_0, \theta))$,
+solving for $y$ at a sequence of times $t_0 < t_1 \leq t_2, \cdots \leq t_n$.
+The residual function $r(y', y, t, \theta)$ will be defined as a function with a certain signature
+and provided along with the initial conditions and output times to one of the
+DAE solver functions.
+
+Similar to DAE solvers, the DAE solver function takes extra arguments
+that are passed along unmodified to the user-supplied system function.
+Because there can be any number of these arguments and they can be of different types,
+they are denoted below as `...`, and the types of these arguments,
+also represented by `...` in the DAE solver call, must match the types of the arguments represented by
+`...` in the user-supplied system function.
+
+### The DAE solver
+
+<!-- array[] vector; dae; (function dae, vector initial_state, vector initial_state_derivative, real initial_time, array[] real times, ...); -->
+\index{{\tt \bfseries dae }!{\tt (function dae, vector initial\_state, vector initial\_state\_derivative, real initial\_time, array[] real times, ...): array[] vector}|hyperpage}
+
+`array[] vector` **`dae`**`(function dae, vector initial_state, vector initial_state_derivative, real initial_time, array[] real times, ...)`<br>\newline
+Solves the DAE system using the backward differentiation formula (BDF)
+method [@serban_user:2021].
+`r since("2.29")`
+
+<!-- array[] vector; dae_tol; (function dae, vector initial_state, vector initial_state_derivative, real initial_time, array[] real times, data real rel_tol, data real abs_tol, int max_num_steps, ...); -->
+\index{{\tt \bfseries dae\_tol }!{\tt (function dae, vector initial\_state, vector initial\_state\_derivative, real initial\_time, array[] real times, data real rel\_tol, data real abs\_tol, int max\_num\_steps, ...): array[] vector}|hyperpage}
+
+`array[] vector` **`dae_tol`**`(function dae, vector initial_state, vector initial_state_derivative, real initial_time, array[] real times, data real rel_tol, data real abs_tol, int max_num_steps, ...)`<br>\newline
+Solves the DAE system for the times provided using the backward differentiation formula (BDF) method with additional control
+parameters for the solver.
+`r since("2.29")`
+
+### DAE system function
+
+The first argument to the DAE solver is the DAE residual
+function. The DAE residual function must have a `vector` return type, and the
+first three arguments must be a `real`, `vector`, and `vector`, in that order. These three
+arguments are followed by the variadic arguments that are passed through from
+the DAE solver function call:
+
+```stan
+  vector dae(real time, vector state, vector state_derivative, ...)
```

Comment:
Because the function itself is called `dae`, we probably want to use a different placeholder here and in the above signatures. Would `residual` make sense as a name, mathematically?